assimilation_code/programs/obs_sequence_tool/obs_sequence_tool.f90

```diff
@@ -642,6 +666,9 @@ program obs_sequence_tool
          obs_out = obs_in
       endif
 
+      ! changing metadata to output precomputed FOs based on namelist
+      call set_FO_metadata(obs_out)
+
```

Comment:
Here's the magic ... one observation at a time ... this is the first one.

Reply:
i like the magic.  

99% of obs aren't going to have any external FOs or if they do, this tool won't want to change what's there. i guess i'd suggest setting some module global flag during the namelist processing that lets you avoid calling this routine if no changes are going to be made. it simplifies the code path if someone is trying to trace through here and doesn't know or care what FO metadata is. something like:

if (deleting_external_FOs) call set_FO_metadata(obs)

Reply:
This is a good idea, chatted with Tim about this. It is a non-user facing change so we can put it on the todo list for the next round of improvements to obs_sequence_tool.